modules/credential/src/main/java/org/eclipse/xpanse/modules/credential/Credential.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.credential;
+
+import org.eclipse.xpanse.modules.credential.enums.CredentialType;
+
+/**
+ * The credential object.
+ */
+public class Credential {
+
+    /**
+     * The name of the credential, this field is provided by the plugins.
+     */
+    String name;

```

Comment:
can we make this class to pack all details together?
1. name -> may be we can even ignore this name field too? 
2. type
3. List of variable names.

Reply:
I meant to make the Credential class to cover all kinds of credentials. Variables should be only one kind of them.
so we can use the Credential at the interface, and safely extend more kinds of Credentials.